packages/experimental/TabList/src/TabList/useTabList.ts

```diff
@@ -53,21 +55,35 @@ export const useTabList = (props: TabListProps): TabListInfo => {
     [setTabKeys],
   );
 
-  const state: TabListState = {
-    context: {
-      addTabKey: addTabKey,
-      appearance: appearance,
-      disabled: disabled,
-      invoked: invoked,
-      onTabSelect: data.onKeySelect,
-      removeTabKey: removeTabKey,
-      selectedKey: selectedKey ?? data.selectedKey,
-      setSelectedTabRef: setSelectedTabRef,
-      setInvoked: setInvoked,
-      size: size,
-      tabKeys: tabKeys,
-      vertical: vertical,
-    },
+  // State variables and functions for saving layout info and other styling information to style the animated indicator.
+  const [listLayoutMap, setListLayoutMap] = React.useState<{ [key: string]: TabLayoutInfo }>({});
+  const [tabListLayout, setTabListLayout] = React.useState<LayoutRectangle>();
+  const [userDefinedAnimatedIndicatorStyles, setUserDefinedAnimatedIndicatorStyles] = React.useState<AnimatedIndicatorStyles>({
+    container: {},
+    indicator: {},
+  });
+
+  const addTabLayout = (tabKey: string, layoutInfo: TabLayoutInfo) => {
+    setListLayoutMap((prev) => ({ ...prev, [tabKey]: { ...prev[tabKey], ...layoutInfo } }));
+  };
+
+  const updateStyles = (update: AnimatedIndicatorStylesUpdate) =>
+    setUserDefinedAnimatedIndicatorStyles((prev) => {
+      const newStyles: AnimatedIndicatorStyles = { ...prev };
```

Comment:
>const newStyles: AnimatedIndicatorStyles = { ...prev }; [](http://example.com/codeflow?start=6&length=55)

Before running setUserDefinedAnimatedIndicatorStyles, we should probably make sure that one of update.container or update.indicator is defined, otherwise this will cause wasted work